sick_scan2 update & release for dashing

Upon building `sick_scan2` node in ROS Dashing, there are several deprecation warnings. It would be great to have this node then released into the ROS Debian buildfarm.

**Warnings:**
```C++
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp: In function ‘int mainGenericLaser(int, char**, std::__cxx11::string)’:
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp:182:58: warning: ‘void rclcpp::Node::set_parameter_if_not_set(const string&, const ParameterT&) [with ParameterT = std::__cxx11::basic_string<char>; std::__cxx11::string = std::__cxx11::basic_string<char>]’ is deprecated: use declare_parameter() instead [-Wdeprecated-declarations]
           node->set_parameter_if_not_set(paramArr[i], val);

                                                          ^
In file included from /opt/ros/dashing/include/rclcpp/node.hpp:1206:0,
                 from /opt/ros/dashing/include/rclcpp/executors/single_threaded_executor.hpp:28,
                 from /opt/ros/dashing/include/rclcpp/executors.hpp:22,
                 from /opt/ros/dashing/include/rclcpp/rclcpp.hpp:144,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common.h:51,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common_tcp.h:45,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_generic_laser.h:8,
                 from /home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp:73:
/opt/ros/dashing/include/rclcpp/node_impl.hpp:306:1: note: declared here
 Node::set_parameter_if_not_set(
 ^~~~
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp:225:52: warning: ‘void rclcpp::Node::set_parameter_if_not_set(const string&, const ParameterT&) [with ParameterT = std::__cxx11::basic_string<char>; std::__cxx11::string = std::__cxx11::basic_string<char>]’ is deprecated: use declare_parameter() instead [-Wdeprecated-declarations]
   node->set_parameter_if_not_set("name",scannerName);

                                                    ^
In file included from /opt/ros/dashing/include/rclcpp/node.hpp:1206:0,
                 from /opt/ros/dashing/include/rclcpp/executors/single_threaded_executor.hpp:28,
                 from /opt/ros/dashing/include/rclcpp/executors.hpp:22,
                 from /opt/ros/dashing/include/rclcpp/rclcpp.hpp:144,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common.h:51,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common_tcp.h:45,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_generic_laser.h:8,
                 from /home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp:73:
/opt/ros/dashing/include/rclcpp/node_impl.hpp:306:1: note: declared here
 Node::set_parameter_if_not_set(
 ^~~~
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp:226:53: warning: ‘void rclcpp::Node::set_parameter_if_not_set(const string&, const ParameterT&) [with ParameterT = std::__cxx11::basic_string<char>; std::__cxx11::string = std::__cxx11::basic_string<char>]’ is deprecated: use declare_parameter() instead [-Wdeprecated-declarations]
   node->set_parameter_if_not_set("hostname",hostname);

                                                     ^
In file included from /opt/ros/dashing/include/rclcpp/node.hpp:1206:0,
                 from /opt/ros/dashing/include/rclcpp/executors/single_threaded_executor.hpp:28,
                 from /opt/ros/dashing/include/rclcpp/executors.hpp:22,
                 from /opt/ros/dashing/include/rclcpp/rclcpp.hpp:144,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common.h:51,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common_tcp.h:45,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_generic_laser.h:8,
                 from /home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp:73:
/opt/ros/dashing/include/rclcpp/node_impl.hpp:306:1: note: declared here
 Node::set_parameter_if_not_set(
 ^~~~
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp:227:45: warning: ‘void rclcpp::Node::set_parameter_if_not_set(const string&, const ParameterT&) [with ParameterT = std::__cxx11::basic_string<char>; std::__cxx11::string = std::__cxx11::basic_string<char>]’ is deprecated: use declare_parameter() instead [-Wdeprecated-declarations]
   node->set_parameter_if_not_set("port",port);

                                             ^
In file included from /opt/ros/dashing/include/rclcpp/node.hpp:1206:0,
                 from /opt/ros/dashing/include/rclcpp/executors/single_threaded_executor.hpp:28,
                 from /opt/ros/dashing/include/rclcpp/executors.hpp:22,
                 from /opt/ros/dashing/include/rclcpp/rclcpp.hpp:144,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common.h:51,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common_tcp.h:45,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_generic_laser.h:8,
                 from /home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_laser.cpp:73:
/opt/ros/dashing/include/rclcpp/node_impl.hpp:306:1: note: declared here
 Node::set_parameter_if_not_set(
 ^~~~
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_scan_common.cpp: In constructor ‘sick_scan::SickScanCommon::SickScanCommon(sick_scan::SickGenericParser*)’:
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_scan_common.cpp:231:59: warning: ‘void rclcpp::Node::set_parameter_if_not_set(const string&, const ParameterT&) [with ParameterT = double; std::__cxx11::string = std::__cxx11::basic_string<char>]’ is deprecated: use declare_parameter() instead [-Wdeprecated-declarations]
    node->set_parameter_if_not_set(PARAM_MIN_ANG, min_angle);
                                                           ^
In file included from /opt/ros/dashing/include/rclcpp/node.hpp:1206:0,
                 from /opt/ros/dashing/include/rclcpp/executors/single_threaded_executor.hpp:28,
                 from /opt/ros/dashing/include/rclcpp/executors.hpp:22,
                 from /opt/ros/dashing/include/rclcpp/rclcpp.hpp:144,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common.h:51,
                 from /home/robo/ros2_ws/src/sick_scan2/driver/src/sick_scan_common.cpp:59:
/opt/ros/dashing/include/rclcpp/node_impl.hpp:306:1: note: declared here
 Node::set_parameter_if_not_set(
 ^~~~
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_scan_common.cpp:232:59: warning: ‘void rclcpp::Node::set_parameter_if_not_set(const string&, const ParameterT&) [with ParameterT = double; std::__cxx11::string = std::__cxx11::basic_string<char>]’ is deprecated: use declare_parameter() instead [-Wdeprecated-declarations]
    node->set_parameter_if_not_set(PARAM_MAX_ANG, max_angle);
                                                           ^
In file included from /opt/ros/dashing/include/rclcpp/node.hpp:1206:0,
                 from /opt/ros/dashing/include/rclcpp/executors/single_threaded_executor.hpp:28,
                 from /opt/ros/dashing/include/rclcpp/executors.hpp:22,
                 from /opt/ros/dashing/include/rclcpp/rclcpp.hpp:144,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common.h:51,
                 from /home/robo/ros2_ws/src/sick_scan2/driver/src/sick_scan_common.cpp:59:
/opt/ros/dashing/include/rclcpp/node_impl.hpp:306:1: note: declared here
 Node::set_parameter_if_not_set(
 ^~~~
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_scan_common.cpp:264:83: warning: ‘std::shared_ptr<PublisherT> rclcpp::Node::create_publisher(const string&, const rmw_qos_profile_t&, std::shared_ptr<_Up>) [with MessageT = sensor_msgs::msg::LaserScan_<std::allocator<void> >; AllocatorT = std::allocator<void>; PublisherT = rclcpp::Publisher<sensor_msgs::msg::LaserScan_<std::allocator<void> > >; std::__cxx11::string = std::__cxx11::basic_string<char>; rmw_qos_profile_t = rmw_qos_profile_t]’ is deprecated: use create_publisher(const std::string &, const rclcpp::QoS &, ...) instead [-Wdeprecated-declarations]
         pub_ = getMainNode()->create_publisher<sensor_msgs::msg::LaserScan>("scan");
                                                                                   ^
In file included from /opt/ros/dashing/include/rclcpp/node.hpp:1206:0,
                 from /opt/ros/dashing/include/rclcpp/executors/single_threaded_executor.hpp:28,
                 from /opt/ros/dashing/include/rclcpp/executors.hpp:22,
                 from /opt/ros/dashing/include/rclcpp/rclcpp.hpp:144,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common.h:51,
                 from /home/robo/ros2_ws/src/sick_scan2/driver/src/sick_scan_common.cpp:59:
/opt/ros/dashing/include/rclcpp/node_impl.hpp:96:1: note: declared here
 Node::create_publisher(
 ^~~~
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_caller.cpp: In function ‘int main(int, char**)’:
/home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_caller.cpp:173:46: warning: ‘void rclcpp::Node::set_parameter_if_not_set(const string&, const ParameterT&) [with ParameterT = int; std::__cxx11::string = std::__cxx11::basic_string<char>]’ is deprecated: use declare_parameter() instead [-Wdeprecated-declarations]
     node->set_parameter_if_not_set("ABC",1000);

                                              ^
In file included from /opt/ros/dashing/include/rclcpp/node.hpp:1206:0,
                 from /opt/ros/dashing/include/rclcpp/executors/single_threaded_executor.hpp:28,
                 from /opt/ros/dashing/include/rclcpp/executors.hpp:22,
                 from /opt/ros/dashing/include/rclcpp/rclcpp.hpp:144,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common.h:51,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_scan_common_tcp.h:45,
                 from /home/robo/ros2_ws/src/sick_scan2/include/sick_scan/sick_generic_laser.h:8,
                 from /home/robo/ros2_ws/src/sick_scan2/driver/src/sick_generic_caller.cpp:75:
/opt/ros/dashing/include/rclcpp/node_impl.hpp:306:1: note: declared here
 Node::set_parameter_if_not_set(
 ^~~~
```

--- a/driver/src/sick_generic_caller.cpp
+++ b/driver/src/sick_generic_caller.cpp
@@ -170,7 +170,7 @@ int main(int argc, char **argv)
     setMainNode(node);
     rclcpp::Logger node_logger = node->get_logger();
 
-    node->set_parameter_if_not_set("ABC",1000);
+    node->declare_parameter("ABC", 1000);
 	char nameId[] = "__name:=";
 	char nameVal[MAX_NAME_LEN] = { 0 };
 	char **argv_tmp; // argv_tmp[0][0] argv_tmp[0] identisch ist zu (*argv_tmp)

--- a/driver/src/sick_generic_laser.cpp
+++ b/driver/src/sick_generic_laser.cpp
@@ -179,7 +179,7 @@ int mainGenericLaser(int argc, char **argv, std::string nodeName)
         callParamName +=  paramArr[i];
         if (tag.compare(callParamName) == 0)
         {
-          node->set_parameter_if_not_set(paramArr[i], val);
+          node->declare_parameter(paramArr[i], val);
           std::string s = "Set param " + paramArr[i] + " to " + val;
           RCLCPP_INFO(node->get_logger(),s.c_str());
 
@@ -222,9 +222,9 @@ int mainGenericLaser(int argc, char **argv, std::string nodeName)
   bool useTCP = true;
 
 
-  node->set_parameter_if_not_set("name",scannerName);
-  node->set_parameter_if_not_set("hostname",hostname);
-  node->set_parameter_if_not_set("port",port);
+  node->declare_parameter("name",scannerName);
+  node->declare_parameter("hostname",hostname);
+  node->declare_parameter("port",port);
 
   int timelimit = 5;
 #if TODO

--- a/driver/src/sick_scan_common.cpp
+++ b/driver/src/sick_scan_common.cpp
@@ -223,18 +223,9 @@ namespace sick_scan
 
 			auto node = getMainNode();
 
-			double min_angle, max_angle, res_angle;
+			double min_angle = node->declare_parameter(PARAM_MIN_ANG, -2.35619449);
+			double max_angle = node->declare_parameter(PARAM_MAX_ANG, +2.35619449);
 
-			min_angle = -2.35619449;
-			max_angle = +2.35619449;
-
-			node->set_parameter_if_not_set(PARAM_MIN_ANG, min_angle);
-			node->set_parameter_if_not_set(PARAM_MAX_ANG, max_angle);
-
-			//			node->set_parameter_if_not_set(PARAM_RES_ANG, );
-
-			node->get_parameter(PARAM_MIN_ANG, min_angle);
-			node->get_parameter(PARAM_MAX_ANG, max_angle);
 			cfg.min_ang = min_angle;
 			cfg.max_ang = max_angle;
 			cfg.frame_id = "laser";
@@ -261,7 +252,7 @@ namespace sick_scan
 		// scan publisher
 		pub_ = nh_.advertise<sensor_msgs::LaserScan>("scan", 1000);
 #endif
-        pub_ = getMainNode()->create_publisher<sensor_msgs::msg::LaserScan>("scan");
+                pub_ = getMainNode()->create_publisher<sensor_msgs::msg::LaserScan>("scan", 10);
 
     }
 

